animation_nodes/nodes/mesh/mesh_points_scatter.py

```diff
@@ -0,0 +1,106 @@
+import bpy
+import random
+from bpy.props import *
+from ... events import propertyChanged
+from ... base_types import AnimationNode
+from . c_utils import trianglePolygonsArea, triangulatePolygons
+from ... algorithms.random import uniformRandomDoubleWithTwoSeeds
+from ... data_structures import LongList, FloatList, Vector3DList, VirtualDoubleList
+
+class MeshPointsScatterNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_MeshPointsScatterNode"
+    bl_label = "Mesh Points Scatter"
+    errorHandlingType = "EXCEPTION"
+
+    nodeSeed: IntProperty(update = propertyChanged)
+
+    triangulate: BoolProperty(name = "Triangulate Mesh", default = False,
+        update = AnimationNode.refresh)
+
+    def setup(self):
+        self.randomizeNodeSeed()
+
+    def create(self):
+        self.newInput("Mesh", "Mesh", "mesh", dataIsModified = True)
+        self.newInput("Integer", "Seed", "seed")
+        self.newInput("Integer", "Amount", "amount", value = 10, minValue = 0)
+        self.newInput("Float List", "Weights", "weights")
+
+        self.newOutput("Vector List", "Points", "points")
+
+    def draw(self, layout):
+        row = layout.row(align = True)
+        row.prop(self, "nodeSeed", text = "Node Seed")
+        row = layout.row(align = True)
+        row.prop(self, "triangulate", text = "Triangulate Mesh")
+
+    def execute(self, mesh, seed, amount, weights):
+        vertices = mesh.vertices
+        polygons = mesh.polygons
+
+        if len(vertices) == 0 or len(polygons) == 0 or amount == 0:
+            return Vector3DList()
+
+        if self.triangulate: polygons = triangulatePolygons(polygons)
+        if polygons.polyLengths.getMaxValue() > 3:
+            self.raiseErrorMessage("Mesh should have triangle polygons.")
+
+        return self.randomPointsScatter(vertices, polygons, seed, amount, weights)
+
+    # Mesh Sampling basis on Barycentric Coordinate System: http://www.joesfer.com/?p=84
+    def randomPointsScatter(self, vertices, polygons, seed, amount, weights):
+        areas = trianglePolygonsArea(vertices, polygons)
+        areaMin = areas.getMinValue()
+        if areaMin == 0: self.raiseErrorMessage("Remove doubles.")
+
+        if len(weights) != 0:
+            usePolyWeight = True
+            polyAmount = len(polygons)
+            polyWeights = FloatList(length = polyAmount)
+            weights = VirtualDoubleList.create(weights, 1).materialize(polyAmount)
+            for i, polygon in enumerate(polygons):
+                polyWeights[i] = (weights[polygon[0]] + weights[polygon[1]] + weights[polygon[2]]) / 3
+        else:
+            usePolyWeight = False
```

Comment:
Weights should always be used. You already use a virtual list with a default of 1, so this should work even if `len(weights) == 0`.

Reply:
Done.